lib/user_comm_screener.rb

```diff
@@ -193,6 +232,20 @@ def load_preference_map
     UserCommPrefs.new(acting_user, resolved_user_communication_preferences)
   end
 
+  def actor_preferences
+    @actor_preferences ||= {
+      muting: @target_users.keys & actor_communication_preferences.filter_map do |pref|
```

Comment:
I don't quite follow why we need to intersect the `target_users` against the actor's preferences when the method which relies on the preferences accepts a `user_id`.

Reply:
Hmm I could probably filter `actor_communication_preferences` by the `@target_users.keys` as well as I do for `user_communication_preferences`, that would get rid of the need to do this intersection (which is done to remove preferences the actor may have that are not related to the target users).